### PR TITLE
Cleanup README, add docs, remove unused deps, upgrade Webpack

### PR DESCRIPTION
**Pull request by kahboom**
## Overview
Not much going on here, just did some cleanup and changes before creating the modules we'll be needing (waiting on confirmation/collaboration of entity relationships).

Changes:
- Cleaned up the `README.md` to be shorter and include only the most relevant information. Updated it to include more important stuff.
- Removed unused dependencies from the `package.json`.
- Upgraded Webpack version for a bug fix.
- Added a `/docs` directory for documentation, and to keep the main `README.md` file concise.

cc @gashcrumb

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,269 +1,158 @@
...
-* What about Webpack 2?
- * If you're looking for Webpack 2 version then see the [experimental version](https://github.com/gdi2290/angular2-webpack2-starter) that will be merged soon.
-* How do I turn on Hot Module Replacement
- * Run `npm run server:dev:hmr`
-
-___
-
-# License
- [MIT](/LICENSE)
+*Hawtio-iPaaS is available for use under the [MIT License](/LICENSE).*
```
Comment (gashcrumb) [inline README.md]: We can get rid of this, we've already a LICENSE file, also it should be Apache 2

File: docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * please see issue [#185](https://github.com/AngularClass/angular2-webpack-starter/issues/185) and PR [196](https://github.com/AngularClass/angular2-webpack-starter/pull/196)
+* How do I include bootstrap or jQuery?
+ * please see issue [#215](https://github.com/AngularClass/angular2-webpack-starter/issues/215) and [#214](https://github.com/AngularClass/angular2-webpack-starter/issues/214#event-511768416)
+* I'm getting an error about not finding my module that I installed?
+ * please see [How to include or create custom type definitions](https://github.com/AngularClass/angular2-webpack-starter/wiki/How-to-include-or-create-custom-type-definitions) and [custom-typings.d.ts](https://github.com/AngularClass/angular2-webpack-starter/blob/master/src/custom-typings.d.ts)
+* How do I async load a component?
+ * see wiki [How-do-I-async-load-a-component-with-AsyncRoute](https://github.com/AngularClass/angular2-webpack-starter/wiki/How-do-I-async-load-a-component-with-AsyncRoute)
+* Error: Cannot find module 'tapable'
+ * Remove `node_modules/` and run `npm cache clean` then `npm install`
+* What about Webpack 2?
```
Comment (gashcrumb) [inline docs/faq.md]: Wonder is this relevant?

Comment (gashcrumb) [inline docs/faq.md]: Actually a few of these FAQ items seem more relevant to the project generator than to hawtio-ipaas.  I wonder if we should prune these and link to the generator's FAQ instead?

File: README.md (hunk 1 above)
Comment (kahboom) [inline README.md]: Good point.

File: docs/faq.md (hunk 2 above)
Comment (kahboom) [inline docs/faq.md]: Good call, I'll just link to angular2-webpack-starter somewhere.

**Pull request merged**